python/cudf/cudf/core/buffer/spillable_buffer.py

```diff
@@ -287,7 +304,11 @@ def get_ptr(self) -> int:
         from cudf.core.buffer.utils import get_spill_lock
 
         spill_lock = get_spill_lock()
-        if spill_lock is None:
+
+        if mode == "internal_write":
+            if spill_lock is None:
+                spill_lock = SpillLock()
```

Comment:
Irrespective of the discussion below, I am not sure that this can be correct. Since the `spill_lock` created here doesn't escape this function, the lifetime of the lock is not long enough to guarantee that the returned pointer is safe for use on the device.

Reply:
Removed this code